src/CoWSwapEthFlow.sol

```diff
@@ -66,8 +66,9 @@ contract CoWSwapEthFlow is
         (bool success, ) = payable(address(wrappedNativeToken)).call{
             value: amount
         }("");
-        // The success value is intentionally disregarded because depositing native tokens with the standard WETH9
-        // contract cannot revert.
+        // The success value is intentionally disregarded. The callback of the standard WETH9 contract has no revert
+        // path in the code, so it could only revert if the internal call runs out of gas. This is not considered a
+        // security risk since a reverting internal call would just mean that calling this function has no effect.
```

Comment:
Is there a possibility to chose a very specific gas value such that the inner call runs out of gas, but the outer call can finish? If the `CALL` is done with `gas()` (i.e. the full remaining gas) my understanding is "no", but I don't know what Solidity does in case of `.call(...)`.

Also, interestingly, you can save almost than 100 gas by doing the equivalent in assembly (this was also the case with optimization enabled for 1M runs):

```solidity
contract Wrap1 {
    IWrappedNativeToken public immutable wrappedNativeToken;

    constructor() payable {
        wrappedNativeToken = IWrappedNativeToken(msg.sender);
    }

    function wrap(uint256 value) external payable {
        (bool success, ) = payable(address(wrappedNativeToken)).call{value: value}("");
        success;
    }
}

contract Wrap2 {
    IWrappedNativeToken public immutable wrappedNativeToken;

    constructor() payable {
        wrappedNativeToken = IWrappedNativeToken(msg.sender);
    }

    function wrap(uint256 value) external payable {
        IWrappedNativeToken to = wrappedNativeToken;
        assembly {
            pop(call(gas(), to, value, 0, 0, 0, 0))
        }
    }
}
```

`Wrap1.wrap` cost `32554` gas while `Wrap2.wrap` costs `32465` gas (at least, this is what Remix told me).

Reply:
> Is there a possibility to chose a very specific gas value such that the inner call runs out of gas, but the outer call can finish?

It seems this is not possible according to the following test. I ran it in `test/e2e/TradingWithCoWSwap.t.sol`, it fails in the last assert. I also tried with your assembly `wrap` and got the same result.

```solidity
    function testFindGasLimit() public {
        vm.deal(address(ethFlow), 1 ether);
        ethFlow.wrap(1);

        uint256 low = 1;
        uint256 high = 1000000;
        while (high - low > 1) {
            uint256 mid = (low + high) / 2;
            try ethFlow.wrap{gas: mid}(1) {
                high = mid;
            } catch {
                low = mid;
            }
        }
        uint256 balanceBefore = weth.balanceOf(address(ethFlow));
        ethFlow.wrap{gas: high}(1);
        assertEq(weth.balanceOf(address(ethFlow)), balanceBefore);
    }
```

But I also don't know if this would be possible in general, which is why I wrote the comment very conservatively.

> Also, interestingly, you can save almost than 100 gas by doing the equivalent in assembly (this was also the case with optimization enabled for 1M runs):

I tried out to see how it compares in our codebase and I see no gas difference for us, which makes sense as we use the optimizer. :tada:

Reply:
My assumption was based on EIP150: https://eips.ethereum.org/EIPS/eip-150

Reply:
> Interesting. For me its not optimizing to the same code shrug.
I tried again and indeed there is a gas saving of 62 gas. Probably the compiler didn't pick up the file changes on time and so it just ran the test on the same files as before.

Reading Adam's link, we should be seeing a point where the internal transaction fails but the main one succeeds (especially in the assembly version, because of the "all but one 64th"). I don't think it's important now and unfortunately can't investigate thoroughly, but in the future I plan to do more research on this and share any findings here.

(In particular, I don't plan to implement Nick's suggestion until further research is done and maybe leave it to a future update, as I have other ideas on how to increase gas efficiency.)